### PR TITLE
Mosviz zoom to slit or object

### LINKED ISSUE #739: [FEAT] Mosviz: have mosviz zoom in on and center the view on the slit when the image is selected
When a table row is selected, instead of just showing the image fully as it is, mosviz should use the wcs in the image to center on the slit and re-scale the view to match the range of the 2d spectrum.  Something like this should extract the boundaries of the 2d spectrum:

```
spcu = mosviz.app.get_viewer('whichever is the 2d viewer').data()[0]
spcu.wcs.calc_footprint() 
# the above is supposed to yield a len-4 skycoord.  When I tested it right now, it didn't work, but that might change when we migrate from spectral-cube?  at any event, an alternative is:
corner1 = spcu.wcs.pixel_to_world(0,0,0)
corner2 = spcu.wcs.pixel_to_world(n,0,0)
corner3 = spcu.wcs.pixel_to_world(0,m,0)
corner4 = spcu.wcs.pixel_to_world(n,m,0)
```
then use those corners to create a bounding "box" for the slit, and convert that to a square-by lengthening the short axis to match the long one.  Then zoom the image into that box using its wcs.

If the image *has no* wcs, the current behavior of just showing the image makes sense (the implicit assumption then is that its a sensible cutout)

### PR DESCRIPTION
**Pull request by rosteen**
### Description

This pull request is to address #739 (and actually is intended to work for NIRISS to zoom/center on object coordinates as well as the slit case). Still in progress, and also includes some fixes/improvements for other things I've run into while working on it (handling missing BUNIT, image parsing in NIRISS parser).

Closes #739

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [ ] Is a milestone set? Milestone is only currently required for PRs related to Imviz MVP.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/configs/mosviz/plugins/parsers.py
Comment (pllim) [inline jdaviz/configs/mosviz/plugins/parsers.py:77]: I wonder if you can take advantage of the Imviz parser that also handles ASDF/GWCS. It does not return stuff as CCDData but you can write a wrapper to do that part.

Comment (rosteen) [reply jdaviz/configs/mosviz/plugins/parsers.py:77]: Likely, but that will be another PR I think. This was just to get the data I had working at all so I can do other things!

Comment (pllim) [reply jdaviz/configs/mosviz/plugins/parsers.py:77]: See #783

**Comment by rosteen** [comment]
I have this working with the most currrent NIRISS and NIRSPEC datasets (see https://github.com/spacetelescope/jwebbinar_prep/blob/main/mos_session/Mosviz_solutions.ipynb for an example of loading the newer NIRSPEC data if you haven't done it yet).

File: notebooks/MosvizNIRISSExample.ipynb
Comment (pllim) [inline notebooks/MosvizNIRISSExample.ipynb:120]: Did you mean to leave the profiler in?

Comment (rosteen) [reply notebooks/MosvizNIRISSExample.ipynb:120]: Frankly I don't think I meant to include either of the notebooks at all. I'll remove them.

File: jdaviz/configs/mosviz/plugins/slit_overlay/slit_overlay.py
Comment (pllim) [inline jdaviz/configs/mosviz/plugins/slit_overlay/slit_overlay.py:23]: Where did this algorithm come from? Are you sure this is correct?

What I dug up over at https://github.com/spacetelescope/jwst/blob/02ca26dd2fcaaf655144949eaf579db2e8d07d11/jwst/assign_wcs/util.py#L922 implies you might need GWCS. Best to confirm with `jwst` people if you are not sure.

File: notebooks/MosvizExample.ipynb
Comment (pllim) [inline notebooks/MosvizExample.ipynb:80]: Replace with Box link that user can access?

File: jdaviz/configs/mosviz/plugins/slit_overlay/slit_overlay.py
Comment (rosteen) [reply jdaviz/configs/mosviz/plugins/slit_overlay/slit_overlay.py:23]: That's a question for @javerbukh

Comment (pllim) [reply jdaviz/configs/mosviz/plugins/slit_overlay/slit_overlay.py:23]: Since this is now separated from UI stuff, this also means we can write unit tests for it. 😉

File: notebooks/MosvizNIRISSExample.ipynb
Comment (pllim) [reply notebooks/MosvizNIRISSExample.ipynb:120]: If you remove any files, please squash them out of commit history too. Thanks!

**Comment by ojustino** [comment]
I agree that these changes work well with the NIRISS example. I skipped the NIRSpec example since it's not ready, but I tried with my own NIRSpec notebook and had no new issues.

The NIRSpec images showed up at their normal field of view before zooming in, while the NIRISS images appeared already zoomed. I don't know if this is within our control, but if it is, I like the "show your work" element of the NIRSpec approach.

**Comment by rosteen** [comment]
Discussing offline, it sounds like we're ok with letting test coverage slip slightly during crunch time leading up to our ~September 7th release. In which case I just need some approvals.

**Review by pllim — APPROVED** [review]
Demo looked good and Jesse already opened follow-up issue on slit placement, so LGTM. Thanks!

File: jdaviz/configs/mosviz/plugins/parsers.py
Comment (pllim) [inline jdaviz/configs/mosviz/plugins/parsers.py:95]: This is not going to work with GWCS but maybe that is future work...

Comment (rosteen) [reply jdaviz/configs/mosviz/plugins/parsers.py:95]: My vote is future work - this wasn't related to this task, I just had to add this to be able to work on this issue because it was broken.

**Comment by rosteen** [comment]
@ojustino I think the difference you're seeing between NIRISS and NIRSPEC is that for NIRISS, the whole image will briefly show the first time a row is clicked before zooming, but after that since they use the same image row clicks will immediately zoom to the object. For NIRSPEC a new image gets loaded for every row so you'll briefly see the whole image every time before it zooms. If that's not a deal breaker for you, would you mind giving this an approval?

**Review by javerbukh — APPROVED** [review]
Looks good! Once you fix the merge conflict this should be good to merge.

**Comment by ojustino** [comment]
I guess I was beaten to the punch, but @rosteen if you like, I can try the functionality a final time after the conflict is resolved as a sanity check.

**Comment by rosteen** [comment]
Oh, didn't even notice that, must have just happened. Thanks.

**Comment by javerbukh** [comment]
Yeah, all this mosviz work means there is a lot of stepping on toes going on 😅

**Comment by rosteen** [comment]
Rebased, should be good to go.

**Comment by rosteen** [comment]
Rebased, should be good to go. I also created an issue to add unit tests for this later so it doesn't get forgotten.

**Pull request merged**
